src/main.rs

```diff
@@ -122,16 +129,34 @@ fn settings() -> io::Result<Settings> {
         .get_matches();
 
     // Read the threshold.
-    let default_threshold = Byte::from_str(DEFAULT_THRESHOLD).unwrap(); // Manually verified safe
+    let default_threshold = Threshold::Absolute(Byte::from_str(DEFAULT_THRESHOLD).unwrap()); // Manually verified safe
     let threshold = matches.value_of(THRESHOLD_OPTION).map_or_else(
         || Ok(default_threshold),
         |threshold| {
-            Byte::from_str(threshold).map_err(|_| {
-                io::Error::new(
-                    io::ErrorKind::InvalidInput,
-                    format!("Invalid threshold {}.", threshold.code_str()),
-                )
-            })
+            if threshold.ends_with("%") {
+                match threshold.strip_suffix("%").unwrap().trim().parse::<f64>() {
```

Comment:
Instead of checking `threshold.ends_with("%")` and then doing `threshold.strip_suffix("%")` (which also implicitly does the check), why not just match on the result of `threshold.strip_suffix("%")`?

Reply:
done!